jc/parsers/curl_head.py

```diff
@@ -0,0 +1,123 @@
+"""jc - JSON Convert `Curl` string parser
+
+Supports output of Curl when run with the HEAD option (-I)
+
+Usage (cli):
+
+    $ curl --head example.org | jc --curl-head
+
+Usage (module):
+
+    import jc
+    result = jc.parse('curl-head', curl_output)
+
+Schema:
+
+Curl output converted to a dictionary
+
+    {
+      "key1":       string,
+      "key2":       string
+    }
+
+Examples:
+
+curl -s -I google.com | python3 -m jc --curl-head | jq
+{
+  "Location": "http://www.google.com/",
+  "Content-Type": "text/html; charset=UTF-8",
+  "Content-Security-Policy-Report-Only": "object-src 'none';base-uri 'self';script-src 'nonce-XXXXXXXX' 'strict-dynamic' 'report-sample' 'unsafe-eval' 'unsafe-inline' https: http:;report-uri https://csp.withgoogle.com/csp/gws/other-hp",
+  "Date": "Wed, 10 Jan 2024 19:12:42 GMT",
+  "Expires": "Fri, 09 Feb 2024 19:12:42 GMT",
+  "Cache-Control": "public, max-age=2592000",
+  "Server": "gws",
+  "Content-Length": "219",
+  "X-XSS-Protection": "0",
+  "X-Frame-Options": "SAMEORIGIN",
+  "HTTP-Version": "1.1",
+  "Status-Code": "301",
+  "Status-Reason": "Moved Permanently"
+}
+"""
+import jc.utils
+import configparser
+import re
+from jc.parsers import kv
+
+
+class info():
+    """Provides parser metadata (version, author, etc.)"""
+    version = '2.0'
+    description = 'curl head parser'
+    author = ''
+    author_email = ''
+    details = 'Using configparser from the python standard library'
+    compatible = ['linux', 'darwin', 'cygwin', 'win32', 'aix', 'freebsd']
+    tags = ['generic', 'file', 'string']
+
+
+__version__ = info.version
+
+
+def _process(proc_data):
+    """
+    Final processing to conform to the schema.
+
+    Parameters:
+
+        proc_data:   (Dictionary) raw structured data to process
+
+    Returns:
+
+        Dictionary representing a Key/Value pair document.
+    """
+    # remove quotation marks from beginning and end of values
+    for key in proc_data:
+        if proc_data[key] is None:
+            proc_data[key] = ''
+
+        elif proc_data[key].startswith('"') and proc_data[key].endswith('"'):
+            proc_data[key] = proc_data[key][1:-1]
+
+        elif proc_data[key].startswith("'") and proc_data[key].endswith("'"):
+            proc_data[key] = proc_data[key][1:-1]
+
```

Comment:
I might also add a new `utils.timestamp` format so we can add `epoch` and `epoch_utc` fields if the `Date`, `Expires`, `Last-Modified`, etc. fields exist.

Reply:
I added the datetime format to `utils.timestamp` in the `dev` branch so we can add the `epoch_utc` timestamps when we are ready. (format `3500`)

Reply:
a) is the idea to add `_epoch` & `_epoch_utc`?

b) to the fields   `"Date"` &  `"Expires"`?

c) like this code snippet?

https://github.com/kellyjonbrazil/jc/blob/140a61702b903918683b4fd0c2861638266e0805/jc/parsers/tune2fs.py#L246-L256

Reply:
continue this in the discussion.